src/Arriba/Arriba.Composition/Composition/Host.cs

```diff
@@ -60,7 +61,19 @@ private static void AddSecureDatabase(this IServiceCollection services)
             var sp = services.BuildServiceProvider();
             var databaseFactory = sp.GetService(typeof(DatabaseFactory)) as DatabaseFactory;
 
-            services.AddSingleton(databaseFactory.GetDatabase());
+            var secureDatabase = databaseFactory.GetDatabase();
+
+            //HACK: As the crawler don't have the AppRoles available at the JWT Token as claims
+            //needed to grant for each role (PermissionScope) the roles name must use the same name as the Scope.
+            secureDatabase.TableNames.ToList().ForEach(table =>
+            {
+                var security = secureDatabase.Security(table);
+                security.Grant(IdentityScope.Group, PermissionScope.Owner.ToString(), PermissionScope.Owner);
+                security.Grant(IdentityScope.Group, PermissionScope.Writer.ToString(), PermissionScope.Writer);
+                security.Grant(IdentityScope.Group, PermissionScope.Reader.ToString(), PermissionScope.Reader);
+            });
+
+            services.AddSingleton(secureDatabase);
```

Comment:
This seems odd to build the serviceProvider, modify stuff, and then continue adding to the services.  Is this okay to do?  I suspect a refactor could be done to make it cleaner

Reply:
I refactored to get an instance from the factory, but also registered the type as I'm not sure that only the factory is in use.